### ISSUE TITLE
How does it compare to the Stream API provided in AbacusUtil

### ISSUE THREAD

**Issue by lihy70** [streamsupport/streamsupport#9]
Here is link:
https://github.com/landawn/AbacusUtil
http://landawn.com/api-docs/index.html

**Comment by stefan-zobel** [comment]
I've never heard of AbacusUtil before. So you should probably ask the author(s) of AbacusUtil how it compares to the other libraries you've asked this question for.

Anyway, my first rough impression (I might be wrong or may have misunderstood something):

AbacusUtil seems to be a very ambitious jumble of very different/unrelated things. In contrast, streamsupport is only concerned with the Java 8 Stream API and nothing else.

I don't see a noteworthy amount of tests (which is no good sign), streamsupport has plenty of them.

With respect to the Stream API: the whole thing seems to depend on Java 8 language features and APIs. I don't know whether this is intended as an extension of (probably not) or a replacement for the Java 8 Stream API. I don't see how this could be used on Java 7 or an Android API < 24. In contrast, the whole idea of streamsupport is to be a backport of the Java 8 Stream API to be used on Java 6 / 7 and Android versions down to API 14.

streamsupport follows the Java 8 API as close as possible. There is not a single method (apart from the obviously necessary bridging methods between j.u. Collections and j8.u. Streams) that is not present in Java 8, has the same Javadoc, has the exact same behavior and is tested using the same tests as employed by OpenJDK.

In short: streamsupport's raison d'etre is to enable ports of Java 8 Stream code to platforms which don't support it without having to learn anything new. Devs who know the Java 8 API can do the port mechanically, without a lot of thinking.

**Comment by stefan-zobel** [comment]
As an aside: I'm not saying that AbacusUtil is not interesting. It just seems to pursue a goal that is very different from streamsupport's intent. So, different goals, different tools (libraries). Nothing wrong with that.
